Add Two Leetcode Examples and Fibonacci Functions

enpuyou opened the pull request.

gkapfham reviewed (approved):
Hello @enpuyou thanks for creating this PR with all of the leetcode functions.

I am glad to approve it in the way that it is currently written.

One way that you could improve the PR is to ensure that all of the functions have a docstring
that follows the same format. (In fact, I am trying to unify this for all of the existing docstrings in the code and then enforcing the standard through a linter configuration).

So, if you could ensure that:

- Each docstring starts with a capital letter
- Each docstring ends with a period for punctuation
- Each docstring clearly explains the purpose of the function

Then this change would be greatly appreciate, as long as you have a little bit of time.

Thanks again for your work on this PR! Your help is greatly appreciated!

enpuyou commented:
Thanks for the reminder, @gkapfham! I have fixed the docstrings for the newly introduced functions. In addition, I also just added some test cases for the two new leetcode modules to ensure the functions are working correctly.

Lancasterwu reviewed (approved):
Thanks for the PR. It looks good!
